sorting/tim_sort.cpp

```diff
@@ -89,8 +91,27 @@ void printArray(int arr[], int n) {
     std::cout << std::endl;
 }
 
+/**
+ * @brief self-test implementation 
+ * @returns void
+ */
+void tests() {
+    // Case: array of length 65
+    constexpr int N = 65;
+    int arr[N];
+
+    std::iota(arr, arr + N, 0);
+    std::reverse(arr, arr + N);
+    assert(!std::is_sorted(arr, arr + N));
+
+    timSort(arr, N);
+    assert(std::is_sorted(arr, arr + N));
+}
+
 // Driver program to test above function
 int main() {
+    tests(); // run self test implementations
+
     int arr[] = {5, 21, 7, 23, 19};
```

Comment:
you could remove the rest of main so only what remains is
```cpp
int main() {
   tests(); // run self test implementations
   return 0;
}
```

Reply:
@realstealthninja Done, added `const` where it is useful